pkg/trace/api/profiles.go

```diff
@@ -202,3 +299,59 @@ func (m *multiTransport) RoundTrip(req *http.Request) (rresp *http.Response, rer
 	}
 	return rresp, rerr
 }
+
+// profilingTransport wraps an *http.Transport to improve connection hygiene after
+// response body copy errors by flushing idle connections and forcing the next
+// outbound request to close instead of reusing a possibly bad connection.
+
+type profilingTransport struct {
+	base           *http.Transport
```

Comment:
nit: since you are wrapping the transport and just adding one extra field, perhaps no need to name it `base`, and just keep `*http.Transport` embedded?

Reply:
thanks